velox/expression/ExprCompiler.cpp

```diff
@@ -302,8 +303,11 @@ ExprPtr compileCall(
         resultType,
         folly::join(", ", inputTypes));
 
+    // Create a shared_ptr as the memory pool is managed by the ExecCtx
+    auto poolPtr = std::shared_ptr<memory::MemoryPool>(
```

Comment:
If we cannot control the life cycle, we should just pass `memory::MemoryPool*` instead of `shared_ptr` to functions instead of faking it.